packages/compass-components/src/components/bson-value.tsx

```diff
@@ -157,6 +157,37 @@ const BinaryValue: React.FunctionComponent<PropsByValueType<'Binary'>> = ({
 
       return { stringifiedValue: `UUID('${uuid}')` };
     }
+    if (value.sub_type === Binary.SUBTYPE_VECTOR) {
+      const vectorType = value.buffer[0];
+      if (vectorType === Binary.VECTOR_TYPE.Int8) {
+        const truncatedSerializedBuffer = truncate(
+          value.toInt8Array().slice(0, 100).toString(),
+          100
+        );
+        return {
+          stringifiedValue: `Binary.fromInt8Array(new Int8Array([${truncatedSerializedBuffer}]))`,
+        };
+      } else if (vectorType === Binary.VECTOR_TYPE.Float32) {
+        const truncatedSerializedBuffer = truncate(
+          [...value.toFloat32Array().slice(0, 100)]
+            // Using a limited precision and removing trailing zeros for better displaying
+            .map((num) => num.toPrecision(8).replace(/\.?0+$/, ''))
```

Comment:
This is the alternative of not adding this

<img width="1207" alt="Screenshot 2025-04-02 at 11 17 03" src="https://github.com/user-attachments/assets/5eed264c-919c-4c14-8a5e-91e61f11c34a" />

Reply:
I see the value but not sure. In mongosh we're not going to apply this transformation as we'd be using inspect on the array directly, so maybe the disconnect can be confusing

Reply:
We did not end up doing the precision thing in mongosh: https://github.com/mongodb-js/mongosh/commit/999208bd397d69a90ec34ceea0fa7af66fc7b4d0 so there would hypothetically be some discrepancy. That said mongosh is more interactive as far as exploring data goes so maybe it would be more justified to have this adjustment here.